src/entities/user/user.controller.ts

```diff
@@ -25,13 +28,53 @@ class UserController implements IController {
     }
 
     initializeRoutes(): void {
-        this.router.get(this.path, this.getSuggestions);
+        this.router.get(`${this.path}/validate_token`, getJWT, this.validateToken);
+        this.router.post(`${this.path}/login`, this.validator.body(valid.UserLoginSchema), this.login);
+        this.router.get(this.path, getJWT, this.getSuggestions);
         this.router.post(this.path, this.validator.body(valid.UserBodySchema), this.createUser);
-        this.router.get(`${this.path}/:id`, this.getUserById);
-        this.router.patch(`${this.path}/:id`, this.validator.body(valid.UserUpdateBodySchema), this.updateUser);
-        this.router.delete(`${this.path}/:id`, this.deleteUser);
+        this.router.get(`${this.path}/:id`, getJWT, this.getUserById);
+        this.router.patch(`${this.path}/:id`, getJWT, this.validator.body(valid.UserUpdateBodySchema), this.updateUser);
+        this.router.delete(`${this.path}/:id`, getJWT, this.deleteUser);
     }
 
+    validateToken = async (
```

Comment:
I don't see any token validation logic in this method, is this intentional?

Reply:
Yes, this is just a test route to see if the authorise middleware is working properly.
I'll remove it as it's not part of the requested routes.